channels/drive/client/drive_main.c

```diff
@@ -592,13 +614,16 @@ static UINT drive_process_irp_query_directory(DRIVE_DEVICE* drive, IRP* irp)
 	if (!drive || !irp || !irp->Complete)
 		return ERROR_INVALID_PARAMETER;
 
+	if (Stream_GetRemainingLength(irp->input) < 28)
```

Comment:
Confused here: UINT32+UINT8+UINT32+23*UINT8 isn't that 32?

Reply:
You're right again